iOS_SDK/OneSignalSDK/UnitTests/InAppMessagingIntegrationTests.m

```diff
@@ -181,6 +183,128 @@ - (void)testDelaysSettingUpTimers {
     XCTAssertTrue(NSTimerOverrider.hasScheduledTimer);
 }
 
+- (void)testMessageWithRedisplay {
+    let limit = 5;
+    let delay = 60;
+    let firstTrigger = [OSTrigger customTriggerWithProperty:@"prop1" withOperator:OSTriggerOperatorTypeExists withValue:nil];
+
+    let message = [OSInAppMessageTestHelper testMessageWithTriggers:@[@[firstTrigger]] withRedisplayLimit:limit delay:@(delay)];
+    
+    //Time interval mock
+    NSDateComponents* comps = [[NSDateComponents alloc]init];
+    comps.year = 2019;
+    comps.month = 6;
+    comps.day = 10;
+    comps.hour = 10;
+    comps.minute = 1;
+
+    NSCalendar* calendar = [NSCalendar currentCalendar];
+    NSDate* date = [calendar dateFromComponents:comps];
+    NSTimeInterval firstInterval = [date timeIntervalSince1970];
+    
+    [self initOneSignalWithInAppMessage:message];
+    [OSMessagingControllerOverrider setMockDateGenerator: ^NSTimeInterval(void) {
+        return firstInterval;
+    }];
+    
+    [OneSignal addTrigger:@"prop1" withValue:@2];
+    
+    // IAM should be shown instantly and be within the messageDisplayQueue
+    XCTAssertTrue(OSMessagingControllerOverrider.messageDisplayQueue.count == 1);
+    
+    // The display should cause an "viewed" API request
+    XCTAssertEqualObjects(OneSignalClientOverrider.lastHTTPRequestType, NSStringFromClass([OSRequestInAppMessageViewed class]));
+    
+    let iamDisplayed = [[OSMessagingControllerOverrider messageDisplayQueue] objectAtIndex:0];
+    XCTAssertTrue(iamDisplayed.displayStats.lastDisplayTime == -1);
+    
+    [OSMessagingControllerOverrider dismissCurrentMessage];
+    [OneSignalClientOverrider reset:self];
+    
+    XCTAssertTrue([[OSMessagingControllerOverrider messagesForRedisplay] objectForKey:message.messageId] != nil);
+    OSInAppMessage *dismissedMessage = [[OSMessagingControllerOverrider messagesForRedisplay] objectForKey:message.messageId];
+    let lastDisplayTime = dismissedMessage.displayStats.lastDisplayTime;
+    XCTAssertEqual(1, dismissedMessage.displayStats.displayQuantity);
+    XCTAssertEqual(firstInterval, lastDisplayTime);
+    
+    XCTAssertTrue(OSMessagingControllerOverrider.messageDisplayQueue.count == 0);
+    
+    comps.minute = 1 + ceil(delay/60); // delay/60 -> minutes
```

Comment:
Why `ceil`, do we have to always round up?

Reply:
no need, i removed it